components/channel_view/channel_view.jsx

```diff
@@ -111,6 +92,9 @@ export default class ChannelView extends React.PureComponent {
             if (dur2 !== -1) {
                 trackEvent('performance', 'team_switch', {duration: Math.round(dur2)});
             }
+            if (this.props.channelIsArchived && !this.props.viewArchivedChannels) {
```

Comment:
If we're checking this here, do we need this same logic elsewhere? There's 3 separate spots in this PR where we're possibly routing the user to town square if they can't view an archived channel

Reply:
Nope, the other places are, when you receive the deletion from the websocket, in that case I want you to do the redirect only if you don't have archivedChannels activated. And when you accept the deletion of the archived channel in the delete channel confirmation modal. In that case, I only want to redirect you to the channel if you are using archived channels, if not, follow the standard process (delete the channel and redirect automatically to town-square).